sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/EventHubEndToEndTests.cs

```diff
@@ -325,7 +331,7 @@ public static void SendEvents_TestHub(int numEvents, string input, [EventHub(Tes
                 }
 
                 // filter for the ID the current test is using
-                if (events[0] == _testId)
+                if (events[0] == _testId && events.Length == s_eventCount)
```

Comment:
I'm afraid this would make the test flaky considering there is no guarantee all events would arrive in the same batch.